src/utils/local_path.ts

```diff
@@ -45,3 +45,34 @@ export function getOutputFilePathAndName(
 	}
 	throw new Error(`The destination is neither a folder nor a file!`);
 }
+
+/**
+ * Resolves the path, verifies its existance and type to conditionally return its dir and file name
+ * @param {string} destOutputPath - the path from where to extract the dir path and name
+ * @returns {FilePathAndName} - the directory where to put the file and the file name (which is undefined when the provided destOutputPath is a directory)
+ */
+export function getOutputFolderPathAndName(
+	destOutputPath: string,
+	fsStatSyncAndPathResolveWrapper = fsStatSyncAndPathResolve
+): FilePathAndName {
+	const resolvedOutputPath = fsStatSyncAndPathResolveWrapper.resolve(destOutputPath);
+	const outputDirname = dirname(resolvedOutputPath);
+	const outputBasename = basename(resolvedOutputPath);
+	try {
+		const outputPathStats = fsStatSyncAndPathResolveWrapper.statSync(resolvedOutputPath);
+		// the destination does exist
+		if (outputPathStats.isDirectory()) {
```

Comment:
Add a comment for the further PRs
```suggestion
		if (outputPathStats.isDirectory()) {
				// TODO: check case sensitivity conflicts here
```